src/main/java/com/minecolonies/coremod/util/Utils.java

```diff
@@ -434,4 +438,55 @@ public static int getFortuneOf(@Nullable final ItemStack tool)
         }
         return fortune;
     }
+    
+    /**
+     * Verifies if there is one tool with an acceptable level
+     * in a worker's inventory
+     *
+     * @param tool   the type of tool needed
+     * @param inventory the worker's inventory
+     * @param hutLevel the worker's hut level
+     * @return true if tool is acceptable
+     */
+    public static boolean hasToolLevel(final String tool, @NotNull final InventoryCitizen inventory, final int hutLevel)
+    {
+        for (int i = 0; i < inventory.getSizeInventory(); i++)
+        {
+            final ItemStack item = inventory.getStackInSlot(i);
+            final int level = Utils.getMiningLevel(item, tool);
+
+            if (Utils.isTool(item, tool) && verifyToolLevel(item, level, hutLevel))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Verifies if an item has an appropriated grade
+     *
+     * @param item   the type of tool needed
+     * @param level  the type of tool needed
+     * @param hutLevel the worker's hut level
+     * @return true if tool is acceptable
+     */
+    public static boolean verifyToolLevel(final ItemStack item, int level, final int hutLevel)
+    {
+        if (item == null || hutLevel > 4)
+        {
+            return true;
+        }
+        else if (item.hasEffect() && hutLevel <= 2 )
+        {
+            return false;
+        }
+        else if (hutLevel >= level)
+        {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 477. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1871)